include/transport/priority_queue.h

```diff
@@ -42,6 +42,10 @@ namespace qtransport {
         };
 
       public:
+        ~priority_queue() {
+            std::lock_guard<std::mutex> _(_mutex);
```

Comment:
Destruct starts from another thread, which caused a race condition on a pop/push that was happening at the same time.

Reply:
I assume this object is held with a shared pointer. Whatever threads are calling into it should have a valid pointer (i.e., the object's destructor has not been called). Do you know what threads? While this might appear to fix it, it concerns me that we're not first terminating those threads before attempting to destroy the priority queue object.

Reply:
I've removed it for now. This change was a fix back before the shutdown with the picoquic transport.  The segfault was caused by the priority queue destructor, but it is a valid question of why was the order not maintained.  I've created issue #112  to work this issue.